src/message/mailbox/parsers.rs

```diff
@@ -0,0 +1,322 @@
+use chumsky::prelude::*;
+
+// Core Rules
+// https://datatracker.ietf.org/doc/html/rfc2234#section-6.1
+
+// CRLF           =  CR LF
+//                        ; Internet standard newline
+pub fn crlf() -> impl Parser<char, Vec<char>, Error = Simple<char>> {
+    just('\r').chain(just('\n'))
```

Comment:
It is related to what I said in this comment https://github.com/lettre/lettre/pull/839#issuecomment-1356306037. The result (parser) is a bit awkward because I started to implement the regular RFC before realizing that sth is not matching. I will clean up a bit and keep only what is relevant.

Reply:
I'll continue the conversation here so that it is threaded. 

Yes, I think your conclusion is correct. We basically need to assume that any byte >127 was properly encoded. You can look at https://github.com/lettre/lettre/pull/844 for reference because I think that I do that part right. IIRC this can only happen in `atom` and `quoted-string` so we just need to habdle it in those cases. Although it probably wouldn't hurt to treat any non-ASCII byte as a "safe" character like `A` or something.

Reply:
Yes, I see. I think I will just replace regular RFC2822 parsers by [RFC5336](https://datatracker.ietf.org/doc/html/rfc5336#section-3.3) parsers (not all, but just the relevant ones like `uAtom` instead of `atom`, or `uMailbox` and it should be enough.

Regarding the `crlf` it is a composite of the `fws` which represents the "Folding white space and comments". I would say it does not *hurt* to keep it in order to keep compatibility with the original RFC, what do you think?

Reply:
I think it is fine to keep it but maybe leave a note that it won't actually be hit because it isn't parsing the raw header but a pre-parsed representation. Or just skip the dead code but leave a note explaining why it doesn't match the RFC. Either would make sense to me.

Reply:
I will go for the first option, thank you!